### DISCUSSION TITLE
smarty 4.x fatal error php tag {php} ?

### DISCUSSION THREAD

**Discussion by polbadman** [smarty-php/smarty#734] [Q&A] [answered]
hi there.
i replase lib/smarty 3.x to lib/smarty 4(last) .but give me fatal error that .dont suport php tag {php}
error:

`Fatal error: Uncaught --> Smarty Compiler: Syntax error in template "file:C:\xampp\htdocs\smarty-master(1)\smarty-master\demo\templates\header.tpl" on line 6 "{php}echo "polad joon";{/php}" unknown tag 'php' <-- thrown in C:\xampp\htdocs\smarty-master(1)\smarty-master\libs\sysplugins\smarty_internal_templatecompilerbase.php on line 6`

In addition, i search in net and see solved  for perstoshop with this: SmartyBC :

`require_once('../libs/SmartyBC.class.php');
$smarty = new SmartyBC();`

but this file "SmartyBC.class.php" have not in lib/?
sorry for bad my EN language.

**Comment by wisskid** [comment]
Please see the changelog, php tags are no longer supported.

**Reply by polbadman** [reply]
Thank you very much. What should be done now?for use php in tpl file.

**Comment by polbadman** [comment]
hi there.
i read changelog:
Dropped support for {php} and {include_php} tags and embedded PHP in templates. Embedded PHP will now be passed through as is.
all right ,now for use like "echo hello;" in tpl file What should be done now?
for example in tpl file:(PHP will now be passed through as is.???)
ago
` {php}echo "hello"{/php},`
now?
 `<?php echo "hello";?>`
or direct : 
`echo "hello";`
or other way?
i use both way (direct&<?php..)but parser smarty dont complie and just like a comment appreance in output buffer.I even changed the file extension to php.
plz help me.

**Reply by wisskid** [reply]
You cannot use PHP in your templates in Smarty 4. So none of your options will work.

If you just wanted to echo "hello", you would just write "hello", but I suppose you want to do something else?

See https://smarty-php.github.io/smarty/getting-started.html for some basic usage examples.

**Reply by polbadman** [reply]
tanX. "but I suppose you want to do something else?" answer:yes i want use php in tmplate for use of my objects  together smarty protifs .but i think .parser smarty this important option that use php in template dont support .
**(PHP will now be passed through as is.???)**. What does this phrase mean?for use  embedded PHP in templates.
Anyway, thank you. And if there is a way to use php in smarty, please help. Thank you again for your answer.

**Comment by wisskid** [comment, marked as answer]
Keep your PHP and your templates separate. That is what template engines like Smarty are for.
So: run all the PHP code you need to run, assign the results to a template variable and then display the template.

**Reply by ernestbuffington** [reply]
Taking the option away to use it is not good. You are trying to control the developer's choices. CSS is cached by the browser and the flykit mod allows people to design themes on the fly. It loads CSS in a PHP file bypassing the cache system for design purposes only. It is enabled with define("FLYKIT", true); and is only used to create themes. How would you suggest loading CSS as a PHP file with Smarty 4?
